From 3b0de305e13b88007b1e414f656ef8de02e92e1f Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Thu, 7 Jul 2016 16:30:32 -0400
Subject: [PATCH 204/260] Fix compiler warning (#1154226)

Coverity threw a compiler warning for an incompatible pointer type when
referencing a union. Converted the argument to pass the exact member of the
union instead of just the union itself.

Related: rhbz#1154226
---
 grub-core/net/drivers/efi/efinet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/net/drivers/efi/efinet.c b/grub-core/net/drivers/efi/efinet.c
index d7befb2ba..ed103df7a 100644
--- a/grub-core/net/drivers/efi/efinet.c
+++ b/grub-core/net/drivers/efi/efinet.c
@@ -415,7 +415,7 @@ grub_efi_net_config_real (grub_efi_handle_t hnd, char **device,
 		     sizeof (hwaddr.mac));
 
 	grub_net_configure_by_dhcpv6_ack (card->name, card, 0,
-					  &hwaddr, &pxe_mode->dhcp_ack,
+					  &hwaddr, &pxe_mode->dhcp_ack.dhcpv6,
 					  1, device, path);
 	grub_dprintf ("efinet", "device: `%s' path: `%s'\n", *device, *path);
       }
-- 
2.13.0

